eZ/Publish/Core/Persistence/Cache/ContentTypeHandler.php

```diff
@@ -349,8 +367,28 @@ public function publish( $typeId )
         // Clear type cache and all identifier cache (as we don't know the identifier)
         $this->cache->clear( 'contentType', $typeId );
         $this->cache->clear( 'contentType', 'identifier' );
+        $this->cache->clear( 'fieldMap' );
 
         // clear content cache
         $this->cache->clear( 'content' );//TIMBER! (possible content changes)
     }
+
+    /**
+     * @see \eZ\Publish\SPI\Persistence\Content\Type\Handler::getFieldMap
+     */
+    public function getFieldMap( $legacy = false )
```

Comment:
\$legacy is not part of interface so should not be here either. anyway we can avoid this? what is the use case?